Services/IotHub/IotHubConnectionStringManager.cs

```diff
@@ -214,13 +217,13 @@ private void ValidateConnectionString(string connectionString)
             }
         }
 
-        private async Task TestIoTHubReadPermissionsAsync(string connectionString)
+        private void TestIoTHubReadPermissionsAsync(string connectionString)
         {
             var registryManager = RegistryManager.CreateFromConnectionString(connectionString);
 
             try
             {
-                await registryManager.GetDevicesAsync(1, CancellationToken.None);
+                this.devicesWrapper.GetDevices(registryManager, this.config.DevicesStorage.DocumentDbCollection, this.config.DevicesStorage.DocumentDbPageSize);
```

Comment:
why do we need devicesWrapper? this can be replaced by:

registryManager.CreateQuery($"select * from {this.config.DevicesStorage.DocumentDbCollection}", this.config.DevicesStorage.DocumentDbPageSize);

Reply:
You should verify that this method passes or fails, depending on the connection string used. As it is this code probably doesn't do anything, and is always passing, did you verify?

Reply:
I don't understand why the DocumentDB pagination is used when working with IoT Hub, and given the context of this PR pagination is not really required

Reply:
this has been changed as per our discussion